frontend/src/pages/Dashboard/Settings.tsx

```diff
@@ -0,0 +1,266 @@
+`;
+
+const SectionHeader = styled.p`
+  font-family: "Rubik";
+  font-size: 26px;
+  font-weight: 700;
+  text-transform: lowercase;
+  word-wrap: break-word;
+  margin: 6vh 0vh 3vh;
+
+  @media (max-width: 900px) {
+    font-size: 30px;
+  }
+`;
+
+const EvenSpacedDiv = styled.div`
+  @media (max-width: 900px) {
+    flex: 0;
+  }
+
+  @media (min-width: 901px) {
+    flex: 1;
+  }
+`;
+
+const StyledPurpleButton = styled(PurpleButton)`
+  flex: 1;
+  width: 100%;
+  margin: 16px 0px;
+  padding: 10px;
+`;
+
+interface NameTextInputProps {
+  isStudent: boolean;
+}
+const NameTextInput = styled(TextInput)`
+  flex: 3;
+  margin-right: ${({ isStudent }: NameTextInputProps) =>
+    isStudent ? "15px" : "0px"};
+`;
+
+const AgeTextInput = styled(TextInput)`
+  flex: 1;
+  margin-left: 5;
+`;
+
+// An example of using a connector
+const connector = connect(
+  (state) => ({
+    user: getCurrentUser(state),
+    error: getDashboardError(state),
+  }),
+  {
+    updateSettings: UpdateUserSettings.request,
+  }
+);
+
+interface SettingsProps {
+  user: User;
+  error?: Error;
+  updateSettings: ({
+    newName,
+    newAge,
+    newEmail,
+    newPassword,
+    currentPassword,
+  }: UserSettings) => void;
+}
+
+const Settings: FunctionComponent<SettingsProps> = ({
+  user,
+  error,
+  updateSettings,
+}): ReactElement => {
+  let [email, setEmail] = useState("");
+  let [password, setPassword] = useState("");
+  let [confirmPassword, setConfirmPassword] = useState("");
+  let [currentPassword, setCurrentPassword] = useState("");
+  let [name, setName] = useState(user.name);
+  let [age, setAge] = useState(user.age);
+  let [errorString, setErrorString] = useState("");
+  let [networkErrorShown, setNetworkErrorShown] = useState(false);
+  let [showSuccessToast, setShowSuccessToast] = useState(false);
+
+  let updateUserSettings = () => {
+    setErrorString("");
+    setShowSuccessToast(false);
+    if (confirmPassword !== password) {
+      setErrorString(
+        "you need to confirm the password with the same password"
+      );
+      return;
+    }
+
+    if (email.length > 0 && currentPassword.length === 0) {
+      setErrorString(
+        "you need to enter your current password to change your email"
+      );
+      return;
+    }
+
+    if (password.length > 0 && currentPassword.length === 0) {
+      setErrorString(
+        "you need to enter your current password to change your password"
+      );
+      return;
+    }
+
+    setNetworkErrorShown(true);
+
+    updateSettings({
+      newName: name === user.name ? undefined : name,
+      newAge: age === user.age ? undefined : age,
+      newEmail: email === "" ? undefined : email,
+      newPassword: password === "" ? undefined : password,
+      currentPassword:
+        currentPassword === "" ? undefined : currentPassword,
+    });
+  };
+
+  useEffect(() => {
+    if (networkErrorShown) {
+      setShowSuccessToast(true);
+    }
+  }, [user]);
+
+  return (
+    <>
```

Comment:
Do we need the wrapper `<></>`? It might be necessary, but hard to tell from the git diff.